core-default/src/main/java/me/aurium/beetle/defaults/task/sync/TimerSleepBehavior.java

```diff
@@ -0,0 +1,91 @@
+package me.aurium.beetle.defaults.task.sync;
+
+import me.aurium.beetle.api.task.futures.SyncTaskerFuture;
+import me.aurium.beetle.api.task.futures.TaskFuture;
+import me.aurium.beetle.api.task.sync.coordination.BlockingBehavior;
+import me.aurium.beetle.api.task.sync.coordination.SyncConstants;
+import me.aurium.beetle.api.task.sync.SyncPulser;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.LockSupport;
+
+public class TimerSleepBehavior implements BlockingBehavior {
+
+    private final int sleepTimer;
+
+    public TimerSleepBehavior(int sleepTimer) {
+        this.sleepTimer = sleepTimer;
+    }
+
+    public TimerSleepBehavior() {
+        this.sleepTimer = 200;
+    }
+
+    @Override
+    public <T> T waitForJoin(SyncTaskerFuture<T> future, SyncPulser pulser) {
+        while (true) {
+            pulser.pulse();
+
+            T result;
+            if (!(result = future.getNow(SyncConstants.CONSTANT())).equals(SyncConstants.CONSTANT())) {
```

Comment:
A NPE will be thrown when the result of the future is null

Reply:
Why so?

Reply:
Oh my god i forgot how java compares null values with equals lmfao 

i have no idea how i forgot that